packages/frontend/src/redux/crossStateSelectors/selectCollectedAvailableForClaimData.js

```diff
@@ -0,0 +1,81 @@
+import BN from 'bn.js';
+import { isEmpty } from 'lodash';
+import { createSelector } from 'reselect';
+
+import { selectValidatorsFarmData } from '../slices/staking';
+import {
+    selectTokensFiatValueUSD,
+    selectTokenWhiteList,
+} from '../slices/tokenFiatValues';
+import { selectAllContractMetadata } from '../slices/tokens';
+
+const collectFarmingData = (...args) => {
+    try {
+        if (args.some((el) => isEmpty(el))) return [];
+
+        const [
+            validatorsFarmData,
+            contractMetadataByContractId,
+            tokenFiatValues,
+            tokenWhitelist,
+        ] = args;
+        const filteredFarms = Object.keys(validatorsFarmData)
+            .reduce((acc, validatorId) => {
+                return [...acc, ...validatorsFarmData[validatorId].farmRewards];
+            }, [])
+            .filter((farm) => farm.active && +farm.balance > 0);
+
+        const collectedBalance = filteredFarms.reduce((acc, farm) => {
+            const tokenId = farm.token_id;
+            if (acc[tokenId]) {
+                return {
+                    ...acc,
+                    [tokenId]: new BN(acc[tokenId].balance)
```

Comment:
The `collectedBalance` is not well calculated here. The bug is whit this line, it should be `acc[tokenId]`.

Also, it seems like the `reduce` can be simplified here, for example, something like this:
```js
const collectedBalance = filteredFarms.reduce((acc, farm) => ({
    ...acc,
    [farm.token_id]: new BN(acc[farm.token_id])
        .add(new BN(farm.balance))
        .toString()
}), {});
```

Reply:
Yes, you are right. I changed the code and use your suggestion. But there is a problem with filtering farms. When I worked with it, the second farm was active. And now all tokens, that have been framed from it, do not sum to an active farm. Maybe
we have to delete this filtering on active farms. Check the suggestion, please https://github.com/near/near-wallet/pull/2528#discussion_r842957198

Reply:
It seems like the problem is with `validatorsFarmData` - if you enter the staking page there is only one validator, if you click on the third validator and get back, the second validator appears in `validatorsFarmData` and the total balance is calculated properly.

Reply:
Resolved this thread. The fix will be in other threads